x/pki/x509/x509.go

```diff
@@ -52,9 +53,53 @@ func DecodeX509Certificate(pemCertificate string) (*X509Certificate, error) {
 		Certificate:    cert,
 	}
 
+	certificate = PatchCertificate(certificate)
+
 	return &certificate, nil
 }
 
+// This function is needed to patch the Issuer/Subject(vid/pid) field of certificate to hex format.
+// https://github.com/zigbee-alliance/distributed-compliance-ledger/issues/270
+func PatchCertificate(certificate X509Certificate) X509Certificate {
+	oldVIDKey := "1.3.6.1.4.1.37244.2.1"
+	oldPIDKey := "1.3.6.1.4.1.37244.2.2"
+
+	newVIDKey := "vid"
+	newPIDKey := "pid"
+
+	issuer := certificate.Issuer
+	issuer = FormatOID(issuer, oldVIDKey, newVIDKey)
+	issuer = FormatOID(issuer, oldPIDKey, newPIDKey)
+
+	subject := certificate.Subject
+	subject = FormatOID(subject, oldVIDKey, newVIDKey)
+	subject = FormatOID(subject, oldPIDKey, newPIDKey)
+
+	certificate.Issuer = issuer
+	certificate.Subject = subject
+
+	return certificate
+}
+
+func FormatOID(header, oldKey, newKey string) string {
+	subjectValues := strings.Split(header, ",")
+
+	integerLengthToHexString := 8
```

Comment:
The variable name is ambiguous. Better to use `hexStringIntegerLength`. 
Consider also adding a comment with a short explanation

Reply:
Done! ✅